--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,7 +37,7 @@
 
 #add extra search paths for libraries and includes
 SET (LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
-SET (LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Directory where lib will install")
+SET (LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE STRING "Directory where lib will install")
 SET (INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "The directory the headers are installed in")
 SET (CMAKE_MODULES_INSTALL_DIR "${CMAKE_ROOT}/Modules" CACHE PATH "The directory to install FindQJSON.cmake to")
 
@@ -56,7 +56,7 @@
                   ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
                   @ONLY)
   INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
-           DESTINATION lib${LIB_SUFFIX}/pkgconfig)
+           DESTINATION lib${LIB_SUFFIX}/${CMAKE_LIBRARY_ARCHITECTURE}/pkgconfig)
 ENDIF (NOT WIN32)
 
 # Include the cmake file needed to use qt4
@@ -89,7 +89,7 @@
                @ONLY)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/qjson-config.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/qjson-config-version.cmake
-        DESTINATION "lib/cmake/qjson")
+        DESTINATION "lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/qjson")
 
 ADD_CUSTOM_TARGET(uninstall
   "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
